File: test-suite/golden-tests/optional-records-py/foo.py
def makePerson(name, age):
    return {"name": name, "age": age}

def getName(p):
    return p["name"]

def getAge(p):
    return p["age"]

def toNull(x):
    return x

def findPerson(name, people):
    for p in people:
        if p["name"] == name:
            return p
    return None

File: test-suite/golden-tests/optional-records-r/foo.R
makePerson <- function(name, age) {
    list(name = name, age = age)
}

getName <- function(p) {
    p$name
}

getAge <- function(p) {
    p$age
}

toNull <- function(x) {
    return(x)
}

findPerson <- function(name, people) {
    for (p in people) {
        if (p$name == name) return(p)
    }
    return(NULL)
}
